### PR TITLE
Remove queue publisher

### PR DESCRIPTION
**Pull request by bakura10**

### CODE REVIEW

File: src/Queue/Queue.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        $this->messages = [];
+    }
+
+    /**
+     * Flush a queue
+     *
+     * @param  bool $async
+     * @return void
+     */
+    private function doFlush(bool $async)
```
Comment (danizord) [inline src/Queue/Queue.php]: default value?

Comment (bakura10) [inline src/Queue/Queue.php]: It's private. The default value is brought by the "flush"

File: src/Queue/Queue.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+            foreach ($messagesToPush as $key => $message) {
+                $messageParameters = [
+                    'Id'           => $key, // Identifier of the message in the batch
+                    'MessageBody'  => json_encode($message['body']),
+                    'DelaySeconds' => $message['options']['delay_seconds'] ?? null
+                ];
+
+                $parameters['Entries'][] = array_filter($messageParameters, function ($value) {
+                    return $value !== null;
```
Comment (danizord) [inline src/Queue/Queue.php]: Checking if parameter is null before adding to messageParameters is not better?

File: src/Queue/QueueInterface.php [hunk 3]
```diff
@@ -16,28 +16,29 @@
...
-     * Set a new queue using an URL
+     * Set a new SQS URL for the queue
      *
-     * @param  string $queue
-     * @param  string $queueUrl
+     * @param  string $url
      * @return void
      */
-    public function setQueue(string $queue, string $queueUrl);
+    public function setQueueUrl(string $url);
```
Comment (danizord) [inline src/Queue/QueueInterface.php]: I'd inject it through constructor instead.

File: src/Queue/Queue.php (hunk 2 above)
Comment (bakura10) [inline src/Queue/Queue.php]: I don't remmeber why I did that. This code is extracted from SlmQueue, but I remember that I had to do this to make it work, so I'm not going to touch it :D.

File: README.md [hunk 4]
```diff
@@ -131,32 +131,48 @@ on this URL.
 
 ### Pushing message
 
-You can push messages by injecting the `QueuePublisher` service into your classes. You need to first add one or more messages,
-then flush the queue. When flushing, the library will make sure to do as few call as possible to SQS (using optimized SQS batch API),
-and to multiple queues:
+You can push messages by injecting a `MessageQueueInterface` object into your classes.
+
+You can create a queue easily, pre-configured, by using the `MessageQueueRepositoryInterface` class. For instance, assuming
```
Comment (danizord) [inline README.md]: InMemoryMessageQueueRepositoryFactory

Comment (bakura10) [inline README.md]: There is an alias.

Comment (bakura10) [inline README.md]: 'aliases' => [
            MessageQueueRepositoryInterface::class => InMemoryMessageQueueRepository::class
        ],

Comment (danizord) [inline README.md]: > using the `MessageQueueRepositoryInterface` class

it's not class :D

**Pull request merged**